core/vdbe/builder.rs

```diff
@@ -183,8 +185,10 @@ impl ProgramBuilder {
     }
 
     pub fn allocate_label(&mut self) -> BranchOffset {
-        self.next_free_label -= 1;
-        BranchOffset::Label(self.next_free_label)
+        let label_n = self.next_free_label;
+        self.next_free_label += 1;
+        self.label_to_resolved_offset.push(None);
```

Comment:
assert here that `label_to_resolved_offset.len() == next_free_label`?

Reply:
I got rid of `next_free_label` entirely and replaced with `let label_n = self.label_to_resolved_offset.len()`